dashboard/app/dsl/script_dsl.rb

```diff
@@ -84,26 +78,30 @@ def lesson_group(key, properties = {})
     if key
       @lesson_groups << {
         key: key,
-        display_name: properties[:display_name]
+        display_name: properties[:display_name],
+        lessons: []
       }.compact
     end
-    @lesson_group = key
   end
 
   def lesson(name, properties = {})
-    if @lesson
-      @lessons << {
-        lesson: @lesson,
-        visible_after: @lesson_visible_after,
-        scriptlevels: @scriptlevels,
+    # For scripts that don't use lesson groups create a blank non-user facing lesson group
+    if name
+      if @lesson_groups.empty?
+        @lesson_groups << {
+          key: nil,
+          display_name: nil,
+          lessons: []
+        }.compact
```

Comment:
this looks like it will remove the key and display_name fields:
```
irb(main):015:0>         {
irb(main):016:1*           key: nil,
irb(main):017:1*           display_name: nil,
irb(main):018:1*           lessons: []
irb(main):019:1>         }.compact
=> {:lessons=>[]}
```
I think this is ok, so just confirming that this is intentional to kind of document that you're intentionally excluding those fields?

Reply:
If it works the same either way, I'd slightly prefer not using `compact` so it's less confusing, and so we always have the same keys in each hash.